commit 9aef94f7964855bfc454f251f7b47e58a7ddd8d5 (refs/changes/53/3553/1)
Author: Orlando Vazquez <ovazquez@gmail.com>
Date:   2018-03-06T16:50:41-08:00 (1 year, 7 months ago)
    
    TRITON-215 CNAPI waitlist code incorrectly using splice method of array objects

diff --git a/lib/models/waitlist.js b/lib/models/waitlist.js
index 0f4132a..f336e4a 100644
--- a/lib/models/waitlist.js
+++ b/lib/models/waitlist.js
@@ -708,7 +708,7 @@ function (ticket, serverqueue, etag, callback) {
         });
     }
 
-    queue.splice(ticketIdx, ticketIdx+1);
+    queue.splice(ticketIdx, 1);
     serverqueue.tickets[key] = queue;
 
     Object.keys(serverqueue.tickets).forEach(function (k) {
@@ -822,7 +822,7 @@ function (ticket, serverqueue, etag, callback) {
         });
     }
 
-    queue.splice(ticketIdx, ticketIdx+1);
+    queue.splice(ticketIdx, 1);
     serverqueue.tickets[key] = queue;
 
     Object.keys(serverqueue.tickets).forEach(function (k) {
@@ -948,7 +948,7 @@ function (ticket, serverqueue, etag, callback) {
         });
     }
 
-    queue.splice(ticketIdx, ticketIdx+1);
+    queue.splice(ticketIdx, 1);
     serverqueue.tickets[key] = queue;
 
     Object.keys(serverqueue.tickets).forEach(function (k) {
